chromadb/server/fastapi/__init__.py

```diff
@@ -106,6 +106,16 @@ async def catch_exceptions_middleware(
         return await call_next(request)
     except ChromaError as e:
         return fastapi_json_response(e)
+    except ValueError as e:
+        return JSONResponse(
+            content={"error": "InvalidArgumentError", "message": str(e)},
+            status_code=400,
+        )
+    except TypeError as e:
+        return JSONResponse(
+            content={"error": "InvalidArgumentError", "message": str(e)},
+            status_code=400,
+        )
```

Comment:
should we transform these error types to a `ChromaError` inside this handler and then allow the normal error flow to handle them?

Reply:
I thought about that, but it felt silly / not helpful to do so, I could go either way - and don't have a principled way of deciding since the change is localized - I.e we'd catch this type, turn it into a chroma error and rethrow. 

If you have a way to decide lmk - I can change it.

Reply:
yeah current method seems fine

Reply:
I'd like to understand this a bit better; how do we know both of these are due to an `InvalidArgumentError` ? Why don't we throw a `ChromaError` that represents an `InvalidArgumentError` further down the stack?

Reply:
ValueError / TypeError 

"A TypeError occurs when an operation or function is applied to an object of inappropriate type. A ValueError occurs when a built-in operation or function receives an argument that has the right type but an inappropriate value"

Map to chromas invalid argument error. Rather than replace ALL instances in our code of ValueError/TypeError I opted to catch them and transmit them to clients as an InvalidArgumentError.

Reply:
The global replace is probably the 'right' thing in the long run (and an easy starter issue); let's make an issue for that. Meantime this looks fine to land.

Reply:
https://github.com/chroma-core/chroma/issues/3026